lib/process-manager.ts

```diff
@@ -460,17 +460,32 @@ export abstract class ProcessManager {
 
 export class QueryProcessManager<T = string, U = string> extends ProcessManager {
 	_query: (input: T) => U | Promise<U>;
+	timeout: number;
 
-	constructor(module: NodeJS.Module, query: (input: T) => U | Promise<U>) {
+	/**
+	 * @param timeout The number of milliseconds to wait before terminating a query. Defaults to 900000 ms (15 minutes).
+	 */
+	constructor(module: NodeJS.Module, query: (input: T) => U | Promise<U>, timeout: number = 15 * 60 * 1000) {
 		super(module);
 		this._query = query;
+		this.timeout = timeout;
 
 		processManagers.push(this);
 	}
-	query(input: T) {
+	async query(input: T) {
 		const process = this.acquire() as QueryProcessWrapper;
-		if (!process) return Promise.resolve(this._query(input));
-		return process.query(input);
+		const timeout = setTimeout(() => {
```

Comment:
I would avoid starting a timeout when `process` doesn't exist. Yes, it will be immediately cleared because there is no `await` for `this._query`, but that doesn't mean we should unnecessarily add onto a queue just to remove it immediately after.

```javascript
		if (!process) return this._query(input);
```

Reply:
Yes, that would be a good plan.